lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1157,28 +1211,38 @@ def has_dict_changed(new_dict, old_dict):
         if value is not None
     )
     for option, value in defined_options.items():
-        if value != old_dict.get(option):
+        old_value = old_dict.get(option)
+        if not value and not old_value:
+            continue
+        if value != old_value:
             return True
     return False
 
 
-def has_list_of_dicts_changed(new_list, old_list):
+def has_list_changed(new_list, old_list):
     """
-    Check two lists of dicts has differences.
+    Check two lists has differences.
     """
     if new_list is None:
         return False
     old_list = old_list or []
     if len(new_list) != len(old_list):
         return True
     for new_item, old_item in zip(new_list, old_list):
-        if has_dict_changed(new_item, old_item):
+        is_same_type = type(new_item) != type(old_item)
+        if is_same_type:
```

Comment:
```suggestion
        if not is_same_type:
```
?

Reply:
Ah yes the variable name is wrong. In that implementation it should read `is_not_same_type`. Cleaner to change `is_same_type = type(new_item) != type(old_item)` to `is_same_type = type(new_item) == type(old_item)` and keep the variable name with your change.